ui/jobs/event_emitter.ts

```diff
@@ -0,0 +1,231 @@
+import EventEmitter from 'eventemitter3';
+
+import logDefinitions from '../../resources/netlog_defs';
+import { addOverlayListener } from '../../resources/overlay_plugin_api';
+import ZoneInfo from '../../resources/zone_info';
+import { EventResponses as OverlayEventResponses, Party } from '../../types/event';
+import { NetFields } from '../../types/net_fields';
+import { ToMatches } from '../../types/net_matches';
+
+import { Player } from './player';
+import { normalizeLogLine } from './utils';
+
+export type PartialFieldMatches<Field extends keyof NetFields> = Partial<
+  ToMatches<NetFields[Field]>
+>;
+
+export interface EventMap {
+  'player': (data: OverlayEventResponses['onPlayerChangedEvent']) => void;
+  // party changed
+  'party': (party: Party[]) => void;
+  // zone changing
+  'zone/change': (id: number, name: string, info?: typeof ZoneInfo[number]) => void;
+  // battle events
+  'battle/in-combat': (info: { game: boolean; act: boolean }) => void;
+  'battle/wipe': () => void;
+  'battle/target': (target?: { name: string; distance: number; effectiveDistance: number }) => void;
+  // triggered when effect gains or loses
+  'effect/gain': (effectId: string, info: Partial<ToMatches<NetFields['GainsEffect']>>) => void;
```

Comment:
I think `Partial<ToMatches<NetFields[string]>>` can be written as `Partial<NetMatches[string]>` or maybe even better use your new `PartialFieldMatches<string>` helper?

Reply:
Yes, I might forget to change them to the new helper type after I added it.